lib/hardware/magnetic_sensor.hpp

```diff
@@ -0,0 +1,20 @@
+#ifndef PET_PASSAGE_LIB_MAGNETIC_SENSOR_H_
+#define PET_PASSAGE_LIB_MAGNETIC_SENSOR_H_
+
+template <unsigned int Pin>
```

Comment:
We can used an unsigned short int for less memory usage, since pin will realistically never be above 15

Reply:
Not sure if it technically matters as they are inlined at compile time but for the sake of keeping things small I am changing to `uint8_t` which is one byte instead of a shorts 2.